File: Sources/macOS/Packages/Yage/Tests/YageTests/EnvironmentBoundsTests.swift
import SwiftUI
import XCTest

@testable import Yage

class EnvironmentBoundsNoSafeAreaTests: XCTestCase {
    let env = World(name: "test", bounds: CGRect(x: 0, y: 50, width: 400, height: 900))

    func testTopBoundProperlyPlaced() {
        let bound = env.children.first { $0.id == Hotspot.topBound.rawValue }
        XCTAssertNotNil(bound)
        XCTAssertEqual(bound?.frame.minX, 0)
        XCTAssertEqual(bound?.frame.maxX, 400)
        XCTAssertEqual(bound?.frame.minY, 0)
        XCTAssertEqual(bound?.frame.maxY, 2)
    }

    func testRightBoundProperlyPlaced() {
        let bound = env.children.first { $0.id == Hotspot.rightBound.rawValue }
        XCTAssertNotNil(bound)
        XCTAssertEqual(bound?.frame.minX, 400)
        XCTAssertEqual(bound?.frame.maxX, 402)
        XCTAssertEqual(bound?.frame.minY, 0)
        XCTAssertEqual(bound?.frame.maxY, 900)
    }

    func testBottomBoundProperlyPlaced() {
        let bound = env.children.first { $0.id == Hotspot.bottomBound.rawValue }
        XCTAssertNotNil(bound)
        XCTAssertEqual(bound?.frame.minX, 0)
        XCTAssertEqual(bound?.frame.maxX, 400)
        XCTAssertEqual(bound?.frame.minY, 900)
        XCTAssertEqual(bound?.frame.maxY, 902)
    }

    func testLeftBoundProperlyPlaced() {
        let bound = env.children.first { $0.id == Hotspot.leftBound.rawValue }
        XCTAssertNotNil(bound)
        XCTAssertEqual(bound?.frame.minX, 0)
        XCTAssertEqual(bound?.frame.maxX, 2)
        XCTAssertEqual(bound?.frame.minY, 0)
        XCTAssertEqual(bound?.frame.maxY, 900)
    }
}

File: Sources/windows/yage/tests/environment_bounds_tests.py
import unittest
from yage.utils.geometry import Rect
from yage.models.hotspots import Hotspot
from yage.models.world import World


class EnvironmentBoundsNoSafeAreaTests(unittest.TestCase):
    def setUp(self):
        self.env = World(name="test", bounds=Rect(
            x=0, y=50, width=400, height=900))

    def test_top_bound_properly_placed(self):
        bound = next((c for c in self.env.children if c.id ==
                     Hotspot.TOP_BOUND.value), None)
        self.assertIsNotNone(bound)
        self.assertEqual(bound.frame.min_x, 0)
        self.assertEqual(bound.frame.max_x, 400)
        self.assertEqual(bound.frame.min_y, 0)
        self.assertEqual(bound.frame.max_y, 2)

    def test_right_bound_properly_placed(self):
        bound = next((c for c in self.env.children if c.id ==
                     Hotspot.RIGHT_BOUND.value), None)
        self.assertIsNotNone(bound)
        self.assertEqual(bound.frame.min_x, 400)
        self.assertEqual(bound.frame.max_x, 402)
        self.assertEqual(bound.frame.min_y, 0)
        self.assertEqual(bound.frame.max_y, 900)

    def test_bottom_bound_properly_placed(self):
        bound = next((c for c in self.env.children if c.id ==
                     Hotspot.BOTTOM_BOUND.value), None)
        self.assertIsNotNone(bound)
        self.assertEqual(bound.frame.min_x, 0)
        self.assertEqual(bound.frame.max_x, 400)
        self.assertEqual(bound.frame.min_y, 900)
        self.assertEqual(bound.frame.max_y, 902)

    def test_left_bound_properly_placed(self):
        bound = next((c for c in self.env.children if c.id ==
                     Hotspot.LEFT_BOUND.value), None)
        self.assertIsNotNone(bound)
        self.assertEqual(bound.frame.min_x, 0)
        self.assertEqual(bound.frame.max_x, 2)
        self.assertEqual(bound.frame.min_y, 0)
        self.assertEqual(bound.frame.max_y, 900)
